core/src/main/java/hudson/model/Job.java

```diff
@@ -316,8 +318,56 @@ public boolean isBuilding() {
      */
     public boolean isLogUpdated() {
         RunT b = getLastBuild();
-        return b!=null && b.isLogUpdated();
-    }    
+        return b != null && b.isLogUpdated() || isCurrentWorkUnit();
```

Comment:
I doubt it's a right place for such check. It affects not only scheduling, but other usages in Jenkins core and outside